### ISSUE TITLE
Docker compose pull error - failed to register layer

### ISSUE THREAD

**Issue by TimDowker** [Ylianst/MeshCentral#4828]
Latest pull with docker compose fails

root@meshcentral:~/meshcentral# dc pull meshcentral
[+] Running 9/13
 ⠹ meshcentral Pulling                                                                                                                                                                                 13.2s
   ⠿ ca7dd9ec2225 Pull complete                                                                                                                                                                         0.4s
   ⠿ 608de902428b Pull complete                                                                                                                                                                         0.5s
   ⠿ 4f4fb700ef54 Pull complete                                                                                                                                                                         0.8s
   ⠿ 5a9dc2cb5c82 Pull complete                                                                                                                                                                         3.6s
   ⠿ d7cb27b02420 Pull complete                                                                                                                                                                         5.2s
   ⠿ 28c5ad0bc7b7 Pull complete                                                                                                                                                                         6.8s
   ⠿ 866c42fd7491 Pull complete                                                                                                                                                                        10.8s
   ⠿ f015ae82a55b Pull complete                                                                                                                                                                        10.8s
   ⠿ 5fa4efbcbdcc Pull complete                                                                                                                                                                        10.9s
   ⠸ 35627c79e3ff Extracting      [==================================================>]  30.82MB/30.82MB                                                                                               12.3s
   ⠸ fde1f78de037 Download complete                                                                                                                                                                    12.3s
   ⠸ 17af578c19a2 Download complete                                                                                                                                                                    12.3s
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /opt/meshcentral/meshcentral/node_modules/array-tools/node_modules/object-tools/.jshintrc: invalid argument

root@meshcentral:~/meshcentral# docker version 
Client: Docker Engine - Community
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 18:01:58 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       3056208
  Built:            Tue Oct 25 17:59:49 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.10
  GitCommit:        770bd0108c32f3fb5c73ae1264f7e503fe7b2661
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

root@meshcentral:~/meshcentral# docker compose version
Docker Compose version v2.12.1

**Comment by sharifm-informatica** [comment]
This works without problems on multiple systems and locations here. Can you confirm it is still happening for you?

**Comment by TimDowker** [comment]
Cornfirmed.  Still an issue with version 1.1.1.  If I switch my docker-compose file to "ghcr.io/ylianst/meshcentral:1.0.97" it pulls no problem but anything after that fails with the same error message as above.

**Comment by TimDowker** [comment]
Ok I was able to spend more time working on this today and I have figured out a possible solution.  In case anybody else comes along for this error I'll leave the solution I found below.

I am running docker in an LXC on a Proxmox server.  The LXC was set to "unprivileged".  This did not work and produced the error above.  Switching it to "privileged" and it works now.

This error report was what helped lead me to the solution.  Apparantly it has something to do with the id mappings between the LXC and host.

https://github.com/moby/moby/issues/43576#issuecomment-1136056638
